jwst/combine_1d/combine1d.py

```diff
@@ -52,12 +52,12 @@ def __init__(self, ms, spec, exptime_key):
         self.flux = spec.spec_table.field("flux").copy()
         self.error = spec.spec_table.field("error").copy()
         try:
-            self.surf_bright = spec.spec_table.field("surf_bright").copy()
+            self.background = spec.spec_table.field("background").copy()
```

Comment:
You can't do this. The combine_1d step's main purpose is for combining 1D spectra of science targets and hence the primary data of interest will be contained in the `flux` and `surf_bright` columns, which will usually already have had background subtraction applied (if it's possible to do background subtraction for the particular mode). This is why many of the regtests are now failing, including the NRS MOS mode, because it uses combine_1d to combine source-like spectra from slits that happen to contain only background signal, and hence all of the data of interest are in the `flux` and `surf_bright` columns, which are now being ignored by this step.

Reply:
Unfortunately this indicates that we need to carefully rethink all the various scenarios of how master_bkg is applied and where the data "live" in each particular case. For cases that use nodded exposures of point sources, the bkg data naturally live in the "background" column of the incoming x1d product. For the case of dedicated bkg exposures, which are naturally treated as extended sources, the extracted data come from the "source" region of the image/cube, which would normally get stored in the "flux" and "surf_bright" columns of the x1d product, but we agreed to also put a copy of the data in the "background" column. But then in the case of a user-supplied spectrum that had its origins who knows where, we can't make any assumptions at all as to what x1d column contains the background signal we need. The assumption till now was that it was in the flux/surf_bright column. The same is true for the NRS MOS case with dedicated bkg slits. The data in those slits is extracted using a single extraction region and hence it naturally gets stored in flux/surf_bright, not background. Oy, what a mess.

Reply:
Here was my thinking. The masterbackground step is now reading in the background column from x1d not the surf_brightness. So I thought I needed to change combine 1d to combine the background columns instead of the surface brightness columns. So I should go back and undo the combine 1d changes. But when combine 1d is called to combine background data how do we handle that ?

Reply:
"But when combine_1d is called to combine background data how do we handle that?"   That's the big $$$ question. Without having spent more than 5-10 mins thinking about it so far, I'm beginning to wonder if we need a param in the master_bkg step that tells it which column to use, which would be user-settable to allow for customizations. Or at the very least it would have to be set by the master_bkg code based on some hierarchy of logic, covering all the use cases mentioned above. It would be a little like the old way of copying background into surf_bright (or vice versa), but instead of actually copying data, we just tell it which column to use.

Reply:
And perhaps a switch/param in combine_1d as well, that tells it to combine the background data, instead of the flux/surf_bright data? Or it could just always combine all of them and then we leave it up to the downstream step (i.e. master_bkg) to decide which one it needs to use.

Reply:
what if in masterbackground we go back and copy the needed information to surface brightness right in the beginning.
I think it would only have to change IFU extended source data - the background information is copied to surface brightness. 
Now the user background is another thing. If I should go back and change it to using the surf brightness column I can and then later after a more formal review we could decide on a clear path forward. Would that cover everything ?